api/hypershift/v1beta1/azure_types.go

```diff
@@ -1,144 +1,177 @@
 package v1beta1
 
+// AzureVMImageType is used to specify the source of the Azure VM boot image.
+// Valid values are ImageID and AzureMarketplace.
+// +kubebuilder:validation:Enum:=ImageID;AzureMarketplace
 type AzureVMImageType string
 
 const (
-	ImageID          AzureVMImageType = "ImageID"
+	// ImageID is the used to specify that an Azure resource ID of a VHD image is used to boot the Azure VMs from.
+	ImageID AzureVMImageType = "ImageID"
+
+	// AzureMarketplace is used to specify the Azure Marketplace image info to use to boot the Azure VMs from.
 	AzureMarketplace AzureVMImageType = "AzureMarketplace"
 )
 
+// AzureNodePoolPlatform is the platform specific configuration for an Azure node pool.
 type AzureNodePoolPlatform struct {
-	// VMSize is the Azure VM instance type to use for the nodes being created in the nodepool.
+	// vmSize is the Azure VM instance type to use for the nodes being created in the nodepool.
```

Comment:
Godocs for fields should start with the lower case version of the name, as this is what users interact with, serialized data